sklearn/ensemble/_gb.py

```diff
@@ -546,12 +397,52 @@ def fit(self, X, y, sample_weight=None, monitor=None):
                 FutureWarning,
             )
 
+        # TODO(1.2): Remove
+        if self.loss == "ls":
+            warnings.warn(
+                "The loss 'ls' was deprecated in v1.0 and "
+                "will be removed in version 1.2. Use 'squared_error'"
+                " which is equivalent.",
+                FutureWarning,
+            )
+        elif self.loss == "lad":
+            warnings.warn(
+                "The loss 'lad' was deprecated in v1.0 and "
+                "will be removed in version 1.2. Use "
+                "'absolute_error' which is equivalent.",
+                FutureWarning,
+            )
+
+        # TODO(1.3): Remove
+        if self.loss == "deviance":
+            warnings.warn(
+                "The loss parameter name 'deviance' was deprecated in v1.1 and will be "
+                "removed in version 1.3. Use the new parameter name 'log_loss' which "
+                "is equivalent.",
+                FutureWarning,
+            )
+            loss_class = (
+                _gb_losses.MultinomialDeviance
+                if len(self.classes_) > 2
+                else _gb_losses.BinomialDeviance
+            )
+        elif self.loss == "log_loss":
+            loss_class = (
+                _gb_losses.MultinomialDeviance
+                if len(self.classes_) > 2
+                else _gb_losses.BinomialDeviance
+            )
+        else:
+            loss_class = _gb_losses.LOSS_FUNCTIONS[self.loss]
+
+        if is_classifier(self):
+            self._loss = loss_class(self.n_classes_)
+        elif self.loss in ("huber", "quantile"):
+            self._loss = loss_class(self.alpha)
+        else:
+            self._loss = loss_class()
```

Comment:
Note that this requires `self.n_classes_` which is set in `self._validate_y` which used to occur before this, but since you've moved it you'll also need to move the `self._validate_y` call up. I'd suggest moving all the initial data validation to before this `self._loss` resolution. See lines 449-466

Reply:
> thanks for another good PR!

Not as good as your reviews though!